src/ui/services/useIdentityFactory.ts

```diff
@@ -3,6 +3,7 @@ import { useMetaMaskWalletInfo } from "./useMetaMask";
 
 export async function useIdentityFactory(web2Provider: CreateIdentityWeb2Provider, nonce: number): Promise<string | undefined> {
     const walletInfo = await useMetaMaskWalletInfo();
+    const message = `Sign this message to generate your ${web2Provider} Semaphore identity with key nonce: ${nonce}.`;
     
-    return await walletInfo?.signer.signMessage(`Sign this message to generate your ${web2Provider} Semaphore identity with key nonce: ${nonce}.`)
+    return walletInfo?.signer.signMessage(message);
```

Comment:
`Signer.signMessage()` is an async API, so we need to keep on adding `await`.
https://docs.ethers.org/v4/api-wallet.html#signer-api

Reply:
No need to await for return. It's redundant.
https://eslint.org/docs/latest/rules/no-return-await